src/vs/workbench/parts/preferences/browser/preferencesService.ts

```diff
@@ -129,13 +125,28 @@ export class PreferencesService extends Disposable implements IPreferencesServic
 			.then(preferencesEditorModel => preferencesEditorModel ? preferencesEditorModel.content : null);
 	}
 
+	disownPreferencesEditorModel(editorModel: IPreferencesEditorModel<any>): void {
+		const uriStr = editorModel.uri.toString();
+		if (uriStr === this.defaultSettingsResource.toString()) {
+			this.defaultPreferencesEditorModelsInUse.set(editorModel.uri, false);
+		} else {
+			this.defaultPreferencesEditorModels.delete(editorModel.uri);
+			editorModel.dispose();
+		}
+	}
```

Comment:
I see you want to cache the model for not to make the search again. But why not cache it in the Search model? May be call SearchModel IPreferencesSearchService and does the caching. Then no model will have exception in disposing.

Reply:
This isn't really related to searching, it's just to not recreate the full prefs model each time a new instance of the preferences editor is created.